File: backend/src/main/java/com/cineverse/api/entities/Screening.java
package com.cineverse.api.entities;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Table(name = "screenings")
@Data
public class Screening {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "movie_id")
    private Movie movie;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    private LocalDateTime startTime;
    private double price;

    /**
     * Este método permite que el Frontend reciba "endTime" en el JSON automáticamente.
     * Al no tener un atributo 'private LocalDateTime endTime', Hibernate no se confunde.
     */
    public LocalDateTime getEndTime() {
        if (this.startTime != null && this.movie != null && this.movie.getDuration() != null) {
            return this.startTime.plusMinutes(this.movie.getDuration());
        }
        return null;
    }
}

File: aplicacion_movil/app/src/main/java/com/example/cineversemobile/model/Screening.kt
package com.example.cineversemobile.model


data class Screening(
    val id: Long,
    val movie: Movie,
    val room: Room,
    val price: Double,
    val startTime: String
)

data class Room(
    val id: Long,
    val name: String,
    val capacity: Int,
    val rowsCount: Int,
    val seatsPerRow: Int
)